### ISSUE TITLE
etcdctl: `alarm list --write-out fields` fails with a segmentation fault

### ISSUE THREAD

**Issue by dlipovetsky** [etcd-io/etcd#11581]
This is a minor issue, since only the `fields` output format is affected.

I can reproduce with v3.3.18 and v3.4.3:

```
$ etcd &>/dev/null &
$ ETCDCTL_API=3 etcdctl version
etcdctl version: 3.3.18
API version: 3.3

$ ETCDCTL_API=3 etcdctl --endpoints http://localhost:2379 --write-out fields alarm list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb62767]

goroutine 1 [running]:
github.com/coreos/etcd/etcdctl/ctlv3/command.(*fieldsPrinter).hdr(0xc00018c480, 0x0)
        /home/ec2-user/go/src/github.com/coreos/etcd/etcdctl/ctlv3/command/printer_fields.go:37 +0x37
github.com/coreos/etcd/etcdctl/ctlv3/command.(*fieldsPrinter).Alarm(0xc00018c480, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/ec2-user/go/src/github.com/coreos/etcd/etcdctl/ctlv3/command/printer_fields.go:176 +0x4d
github.com/coreos/etcd/etcdctl/ctlv3/command.alarmListCommandFunc(0xc000280e00, 0xc0001c0680, 0x0, 0x4)
        /home/ec2-user/go/src/github.com/coreos/etcd/etcdctl/ctlv3/command/alarm_command.go:80 +0x16a
github.com/coreos/etcd/vendor/github.com/spf13/cobra.(*Command).execute(0xc000280e00, 0xc0001c0600, 0x4, 0x4, 0xc000280e00, 0xc0001c0600)
        /home/ec2-user/go/src/github.com/coreos/etcd/vendor/github.com/spf13/cobra/command.go:572 +0x219
github.com/coreos/etcd/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1521b40, 0xb, 0xc0000d0be8, 0xb84701)
        /home/ec2-user/go/src/github.com/coreos/etcd/vendor/github.com/spf13/cobra/command.go:662 +0x2f4
github.com/coreos/etcd/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /home/ec2-user/go/src/github.com/coreos/etcd/vendor/github.com/spf13/cobra/command.go:618
github.com/coreos/etcd/etcdctl/ctlv3.Start()
        /home/ec2-user/go/src/github.com/coreos/etcd/etcdctl/ctlv3/ctl_nocov.go:25 +0x81
main.main()
        /home/ec2-user/go/src/github.com/coreos/etcd/etcdctl/main.go:40 +0x14d

```

```
$ etcd &>/dev/null &
$ etcdctl version
etcdctl version: 3.4.3
API version: 3.4

$ ./etcdctl --endpoints http://localhost:2379 --write-out fields alarm list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb5ddc7]

goroutine 1 [running]:
go.etcd.io/etcd/etcdctl/ctlv3/command.(*fieldsPrinter).hdr(0xc00021a760, 0x0)
        /tmp/etcd-release-3.4.3/etcd/release/etcd/etcdctl/ctlv3/command/printer_fields.go:38 +0x37
go.etcd.io/etcd/etcdctl/ctlv3/command.(*fieldsPrinter).Alarm(0xc00021a760, 0x0, 0x0, 0x0, 0x0)
        /tmp/etcd-release-3.4.3/etcd/release/etcd/etcdctl/ctlv3/command/printer_fields.go:181 +0x4d
go.etcd.io/etcd/etcdctl/ctlv3/command.alarmListCommandFunc(0xc000251900, 0xc000248bc0, 0x0, 0x4)
        /tmp/etcd-release-3.4.3/etcd/release/etcd/etcdctl/ctlv3/command/alarm_command.go:80 +0x115
github.com/spf13/cobra.(*Command).execute(0xc000251900, 0xc000248b00, 0x4, 0x4, 0xc000251900, 0xc000248b00)
        /home/ec2-user/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0x14b4240, 0xb, 0x14ddec0, 0xb7e200)
        /home/ec2-user/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ec
github.com/spf13/cobra.(*Command).Execute(...)
        /home/ec2-user/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
go.etcd.io/etcd/etcdctl/ctlv3.Start()
        /tmp/etcd-release-3.4.3/etcd/release/etcd/etcdctl/ctlv3/ctl_nocov.go:25 +0x81
main.main()
        /tmp/etcd-release-3.4.3/etcd/release/etcd/etcdctl/main.go:35 +0x77
```

I should be able to take this up, but not immediately.

**Comment by jingyih** [comment]
Thanks for reporting! This looks like a bug. Please feel free to send a fix:)

**Comment by dlipovetsky** [comment]
I was able to reproduce this with a debugger.

When there are no alarms, `out.Header` (type `ResponseHeader`) remains `nil` when this call succeeds:
https://github.com/etcd-io/etcd/blob/1cef112a7979d077746d657bdbec38841d9ae7e2/etcdserver/etcdserverpb/rpc.pb.go#L4250

Is it valid for a ResponseHeader to be `nil`? I thought it would be `nil` only if `grpc.Invoke` returns an error.

**Comment by jingyih** [comment]
> Is it valid for a ResponseHeader to be nil?

Good question. I did not find much documentation on response header. I would assume server should fill the response header.

**Comment by dlipovetsky** [comment]
I debugged the server side a bit.

On the server side, this call returns a response with a `nil` Header field:
https://github.com/etcd-io/etcd/blob/071e70cdc4dfacbcfab6a3f4e92659b3bccaabd2/etcdserver/v3_server.go#L379

The request `MemberID` field is set to `0` (the request is created here: https://github.com/etcd-io/etcd/blob/34bd797e6754911ee540e8c87f708f88ffe89f37/clientv3/maintenance.go#L108), and I wondered whether that had to do with the response Header being nil. I tried a different value (the server's own member ID), but the Header remained nil.
